Cargo.toml

```diff
@@ -10,6 +10,9 @@ documentation = "https://docs.rs/redis"
 license = "BSD-3-Clause"
 edition = "2018"
 
+[workspace]
```

Comment:
I tend to find nested crate workspaces confusing to navigate. Would you mind changing this so that the top-level `Cargo.toml` only contains the workspace, and move the redis crate into a `redis` directory?

Reply:
Should that be done in a separate PR?

Reply:
(to avoid confusing commit history)

Reply:
In my mind the ideal way to submit it would be a single PR with clean commits: one commit that moves the code around and one commit that adds the new crate. Each commit should ideally pass CI, though I usually don't test that explicitly (but it sucks if formatting changes for one commit make it into another).

Reply:
So it sounds like that PRs are not squashed before merge then. Will do as separate commits in this PR then. (Context: My company merges via squash-and-merge generally for PRs so a separate PR would be the only way to accomplish having separate commit history in that case.)

Reply:
I'm also okay with separate PRs if you prefer. For my own changes I usually submit clean commits "manually" (squashing with interactive rebase).